app/api/speaker_image_sizes.py

```diff
@@ -4,15 +4,29 @@
 from app.api.schema.image_sizes import SpeakerImageSizeSchema
 from app.models import db
 from app.models.image_size import ImageSizes
-
+from app.api.helpers import db as db_helper
 
 class SpeakerImageSizeDetail(ResourceDetail):
     """
-    Speaker Image_size detail by id
+    Speaker Image_size detail by type
     """
+    @classmethod
+    def before_patch(self, args, kwargs, data=None):
+        # overrides the patch request id by image size type
+        speaker_image_size, is_created = db_helper.get_or_create(ImageSizes, type='speaker-image')
+        if is_created:
+            speaker_image_size.init_speaker_image_size()
+            db.session.commit()
+        kwargs['id'] = speaker_image_size.id
+
     @classmethod
     def before_get(self, args, kwargs):
-        kwargs['id'] = 2
+        speaker_image_size, is_created = db_helper.get_or_create(ImageSizes, type='speaker-image')
+        if is_created:
+            speaker_image_size.init_speaker_image_size()
+            db.session.commit()
+        kwargs['id'] = speaker_image_size.id
+
     decorators = (api.has_permission('is_admin', methods="PATCH", id="2"),)
```

Comment:
This is fetching id statically

Reply:
I changed the id in the before_patch function

Reply:
So remove 2 from here. What's its use other than to confuse people

Reply:
@iamareebjamal ? By removing the line
`decorators = (api.has_permission('is_admin', methods="PATCH", id="2"),)`

My endpoint of PATCH fails with error
`  File "/usr/local/bin/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 297, in patch
    if json_data['data']['id'] != str(kwargs[self.data_layer.get('url_field', 'id')]):`

Reply:
Then it is already doing something wrong and should be fixed. This is fetching the ID of 2 unnecessarily

Reply:
@iamareebjamal - The issue is similar to 
https://github.com/fossasia/open-event-server/issues/5118

The permission manager is invoked in the `has_permission` method which provides the id for the above issue
https://github.com/fossasia/open-event-server/blob/3ea38141a731cf51a3c8314e02b5ec38d0b8e932/app/api/helpers/permission_manager.py#L394-L400

We also created an issue here: #6708 

The issue was also found some 1 month ago with id of orders - of @prateekj117 PR for creating order and managing permission.

Reply:
It shouldn't fetch anything

Reply:
@iamareebjamal The patch method requires id of the instance to be patch, and our end point doesn't have id that's why it's not working imo. How it should be fixed now please guide me?

Reply:
@iamareebjamal - I need your help here, thanks

Reply:
@iamareebjamal - Any inputs sir for this?